### PR TITLE
[READY] [MODULAR] Blueshift wakes up after taking "Worst Nap Ever" [MDB IGNORE]

### PR DESCRIPTION
**Pull request by W4rd3nn**
## About The Pull Request
Re-opens #20655 

Blueshift is a pretty neat map and there's a lot of work put into it. I wanna help make it a reality.

<details>

<summary>Original PR Body</summary>

> After a lot of work and being lazy i finally finished up the revamp of blueshift, it's got it's ass chopped off and is now only 2zs tall!
> With most of the original ship preserved to the best of my ability you will still see a few overhauled sections, namely cargo and engineering which have been made from the ground up to fit the new formfactor.
> Some other departments have had some improvements implemented to better suit players with roleplay opportunities specifically in mind.
> Most notably is the size down of maintenance areas which are now far more streamlined and logical, with a pair of large chunks of maints jungle still up yet harder to reach up north on layer 2, this should give security far less of a painful time searching around the more shady parts of the ship.
> Less interestingly i have taken some special consideration to atmospherics performance by dividing up large areas a little more, which will ensure things are far less performance intensive once there are a few breaches.
> 
> On top of that a long lost project i once had for blueshift has been revitalized and implemented, a whiteship custom made for blueshift which will now hopefully spawn alongside it's parent map.
> This sizable frigate appears to have been lost after making a SOS call including some data of alien lifeforms on board.. who knows what it might be but be careful when exploring the derelict vessel.
> It's cargo bays were laden with goodies when it was in transit so whatever the risk may be it'll be worth it.
> 
> I'm sorry to those who had to wait so long for this map to return, i was kinda demotivated for a while but i'm glad i got around to publishing it again, i love my little spaceship map and i hope you guys do too, i'll be happy to hear feedback when you have it.

</details>

## How This Contributes To The Skyrat Roleplay Experience
I'm evil I wanna make more work for maptainers >:)

But in all seriousness, I think blueshift has a unique position to offer "round-start RP opportunities". Meaning that you don't need a lot of setup to do a gimmick. Wanna run a convenience store? Just go hop behind the counter, no setup needed. Maint bar? Go nuts, there's already a nightclub and some silly maint areas. 

That being said, the map is still built for people who would like to change things up. There's lots of space to build, edit, and remove areas. Just for those who don't want to spend 30 minutes to an hour setting up their little locational gimmick, they can jump right in.

## Proof of Testing

<details>

<summary>Screenshots/Videos</summary>

Z1

Z2

</details>

## Changelog

:cl:IsaacTheSharkWolf & Motho
add: The derelict NSV Blueshift has been located and refitted for active duty after being "lost with all hands" for nearly a year. NanoTrasen leadership has declined to comment on why the ship went missing in the first place, sparking rumors of esoteric circumstances regarding the ship's disappearance.
config: Revamped Blueshift added back into rotation.
/:cl:

### CODE REVIEW

**Comment by SylvetteSylph** [comment]
WE ARE SO BACK

**Comment by MidoriWroth** [comment]
thank you...  we are so blessed... Go before us, O Lord, in all our doings with your most gracious favor, and further us with your continual help; that in all our works begun, continued, and ended in you, we may glorify your holy Name, and finally, through your mercy, obtain everlasting life; through Jesus Christ our Lord.

**Comment by W4rd3nn** [comment]
Gonna continue on this this weekend :3

**Comment by W4rd3nn** [comment]
I'm gonna fucking lose it if I push this next commit and it finds another fucking S somewhere

**Comment by W4rd3nn** [comment]
I'm.. So... Happy...

**Comment by Draganfrukts** [comment]
image

**Comment by W4rd3nn** [comment]
This is ready for review, BTW.

File: modular_skyrat/modules/mapping/blueshift/code/areas.dm
Comment (Jolly-66) [inline modular_skyrat/modules/mapping/blueshift/code/areas.dm]: Please put this in `**/modules/mapping/area/**`.
I want to get away from maps having their own modules because its needlessly messy and stupid.

File: modular_skyrat/modules/mapping/blueshift/readme.md
Comment (Jolly-66) [inline modular_skyrat/modules/mapping/blueshift/readme.md]: Ditto get rid of this.

**Review by Jolly-66 — CHANGES_REQUESTED** [review]

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
Please let me preface with this: I'll be dumping my old reviews here. I don't know how much has changed since this PR and map changed hands. I'll be looking at the map in between and adjusting my reviews where needed. Some information may be outdated and/or incorrect. 

For the time being, I'll be restarting the entire review, starting with Medical and Service. I'll be tweaking stuff as I go along.

<details>
<summary> Medical Review </summary>

Keep wall mounts to one item, even if they're different dir/x.
image

I know this is stylistic, but please revert these tables back to regular glass tables or reinforced.
image

Ditto.
image

From my previous review, I asked for doors to be removed. You can readd them here, as this'll prevent people from needlessly getting deeper into medbay then they have to be.
image

Cryo still needs to be tweaked. I suggest pushing the main wall and door up, while having the little bit be hall exposed as a "boo-boo treatment center".
image

Also, cryo needs to be reworked (the piping). Please look at another station map to see what I'm talking about.

Consider using the `/isolation/` subtype area for the top here, or make a new area and call it something else. I'll judge whatever you propose.
image

These walls are internal and do not need to be r-walls.
image

Bad placement and this looks bad. Just remove it.
image

This cabinet needs to be removed.
image

This bulletin board is var edited to dir 1 (lol), replace it please.
image

Swap these back over to regular beds (you also forgot to rotate one :()
image

I would consider just making this room normal temp and throw in some flavor thermo unit. Cold areas like this create atmos issues since it has critical items in it. More importantly, people are going to get burned from the cold. The room itself is fine, just the turfs need to be changed to not be cold room turfs (also make sure to remove the air alarm helper)
image

Rotate the laptop.
image

Remove the sec consoles, add a crew monitoring console.
image

Put an abandoned helper on this airlock.
image


</details>

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
<details>
<summary> Service Review </summary>

Please use the bar backroom define here.
image

Get rid of that chem master board. You botanist mains I swear to absolute God.
image
Also plumbing terminates in here, please fix that up (there is no sink to terminate into)

The tools can stay, however, only two can stay and they need to be the random helper. Remove the toolbelt.
image

Nitpick but these shutters should have their own button to be controlled by. They're currently controlled by the north button, which also controls the side shutters.
image

Arcade still needs to be shrunk down. My suggestion, remove the blue carpeted area and give it back to as a little lounge area, like what already exists in front of the arcade. Make it a cafe if it works.
image

These should be their own area defines. Something along the lines with the wrestling arena (bonus points if you name them goofy but keep it wrestling themed).
image

Put shutters on the counter here, closed preferably, like with the kitchen just north east.
image

**The dorms issue**.
You could probably delete the new area defines you wrote and instead use North Stars apartments. For what doesn't exist, make more of. Apartments fit better than dorms, since they're actual rooms, not little hobbit holes.

Sauna still needs it own area define.
image

Barber areas still need to be split up into the sauna, massage parlor respectively.
image

The chimichanga on the floor is funny, but it'll spoil and turn into rot over time. Look in code and find the var that would prevent this from happening, and make sure to MAKE A SUBTYPE please. 
image

Give this a unique area.
image


</details>

File: modular_skyrat/modules/mapping/blueshift/code/areas.dm
Comment (W4rd3nn) [reply modular_skyrat/modules/mapping/blueshift/code/areas.dm]: Moved

File: modular_skyrat/modules/mapping/blueshift/readme.md
Comment (W4rd3nn) [reply modular_skyrat/modules/mapping/blueshift/readme.md]: Removed

**Comment by W4rd3nn** [comment]
> From my previous review, I asked for doors to be removed. You can readd them here, as this'll prevent people from needlessly getting deeper into medbay then they have to be.

Feels hard to justify doors here. I've added them, but pushed back into a separate hallway, making the psych, asylum, and CMO's office behind the doors rather than the recovery ward and virology.

**Comment by W4rd3nn** [comment]
REVIEWS COMPLETED >:)

**Comment by Jolly-66** [comment]
> > From my previous review, I asked for doors to be removed. You can readd them here, as this'll prevent people from needlessly getting deeper into medbay then they have to be.
> 
> Feels hard to justify doors here. I've added them, but pushed back into a separate hallway, making the psych, asylum, and CMO's office behind the doors rather than the recovery ward and virology.

For the record we have AA helpers, which are those small green chevrons. They allow unrestricted access on that side of the door only.

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
<details>
<summary> Minor things </summary>

The clock is now floating. If you can't find a wall don't stress over it.
image

Rotate these beds.
image

Fix the area.
image

I didn't realize at the time, but this button needs to be moved over to a directional helper. Just copy over the vars.
image

Put some windows here to open up the arcade a little more. Also swap out the cabinets to use random helpers.
image

If you're going to make this entire area the sauna, you'll need to readjust the door so it makes sense. As it stands, someone will believe the entire area is under another (I know this sounds confusing, because it is). The best way to think of it is, if you enter a door, you should be in room X, not Y. Poke me if you still don't understand.
image

Bring back the chimichanga and put it somewhere :(
image

</details>

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
<details>
<summary> Science Review </summary>

For geneticts, put windows here instead of panes.
image

Remove this. It's an ancient botany relic.
image

This section of robo needs to be reworked. It's mostly 1x2 walkways, and with the fact we usually have fully staffed robo teams, it can be agonizing to navigate these areas.
image

This entire exp lab area needs to be broken up. You'll most certainly need to make new areas.
image
Also, while you're there, remove the chemistry stuff from the top lab portion. You can probably refit that entire lab to be a vacant lab or destroyed lab. Up to you.

Remove the sec specific consoles here, replace them with any sci applicable consoles (except research).
image

The circuits lab should be moved into their own little area, instead of being next to RnD. Could go into that big exp lab glob.
image

Same deal with CMO, this closet and its contents need to go.
image

This is an old tele-sci room, which can be made into an abandoned maint room. Use an abandoned door helper on the door, reshuffle the loot (remove somethings, change somethings to helpers). Rip up the floors ect.
image

Two things here:
Top portion: Convert those walls into r-walls (excluding door).
Side portion: Convert those walls into regular walls.
image

RD, wheres the Ordnance testing site?
image

Missing ordnance pipes.
image

XB is a fucking massive area, trim it a bit (preferably, the top portion).
image

Also, the cytology chemistry equipment needs to go.
image

Ditto.
image

Waste release button should have an access requirement tired to it.
image

I have words with Ordnance and XB being large/needing some adjusting. However, I believe that can be done later if needed.

</details>

<details>
<summary> The surrounding maints </summary>

This DNA injector has got to go.
image

Ditto with this acid pool. I don't want a pool here period of anything.
image


</details>

**Comment by W4rd3nn** [comment]
Fucked up about the acid death pool :(

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
<details>
<summary> Additional stuff </summary>

Put shutters on these windows + make sure the area for genetics covers them. 
image

This area can't exist being cut off like this. If people get trapped in the experimentor room, they'll be confused as to why. Inch circuits up if you have to (the area).
image

If you really want to give sci a place to do meds job, remove the tray, and add a table and computer frame. Remember, we don't want to give job X job Ys content, otherwise no one will go to job Y when job X can do it.
image

Few things:
Remove the access helpers from the command door, and the door across from it.
Remove the circuit boards.
Remove the toolboxes, swap them out for 2/3 random tool spawners.
Remove the launch pad computer.
image

Please make sure to follow the format of the other area icon sprites (yes, I'm going to be picky with these sprites).
image

Still missing pipes.
image



</details>

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
<details>
<summary> Engineering Review </summary>

Per tradition, remove the security consoles. Add engineering ones (power monitoring, the canary one, whatever)
image

Tone down the MODsuits to only two.
image

Move the wires so they're connected to the emitters (or move said emitters back)
image

Need a wall here, remove the emitter.
image

These are not proper plumbed tanks for plumbing.
image

As an aside, I'm real iffy with the atmos tanks being 2x3, but we'll see if they hold up in a round of SR when we reach that bridge.

</details>

**Review by Jolly-66 — DISMISSED** [review]
<details>
<summary> Cargo review </summary>

Make sure you remove the sec consoles once again. Swap them for cargo stuff.
image

Wall mount overlap :(
image

image

QM perimeter walls need to be r-walls.

image

THE RECYCLER IS NOW DIRECTIONAL!!
Please make sure your branch is 1:1 with master, as recyclers are now directional.
Similarly, I want you to test the mailing system to make sure it works.
image

This can't exist as the mass driver. People will get stuck Also, this thing is airless.
image

No cargo drone bay?
image

Split up the office area, use the lobby.
image

</details>

**Comment by Jolly-66** [comment]
Something I wanna note thats a trend in the reviews:

- Check all remaining Heads of Staff for cabinets with clothing. Remove them.
- Check all remaining guard outposts for sec consoles. Replace them with department specific consoles.
- Ensure r-wall coverage is consistent for rooms.

**Comment by W4rd3nn** [comment]
> Please make sure to follow the format of the other area icon sprites (yes, I'm going to be picky with these sprites).

Funnily enough, this was actually just the wrong area define. This was OUR xenoarch/cytology thingy. I changed it over to TG's cytology area.

**Comment by W4rd3nn** [comment]
> These are not proper plumbed tanks for plumbing.

Not intended to be. These are for fire extinguishers. The plumbing is serviced from the stationary tanks in maint to the east.

**Comment by W4rd3nn** [comment]
reviews compwete btw >:3

**Review by KathrinBailey — DISMISSED** [review]
Security / Command stuff:

# Security
## Sec Medical
• Remove most of this area as per #21680, leave only a few beds and first aid kits for patching yourself up. The rest of this space should be converted to maint. Do not place operating tables/stasis beds here, injured security officers should be taken to medbay for serious injuries.
• As for size, Jolly said 'Broom closet', so first aid area.

## Detective's Office
• Add curtains or pre-open shutters. On other maps, detective's have some sort of privacy shutter so they can noir with their clients without people listening in. Jolly prefers shutters.

## Prison
• Remove the deep fryer as it is completely un-used in food code, it's more of a meme tool used to delete items.
• Food processor on a table as per recent TG kitchen updates.
• Move griddle to where the deep fryer used to be and plop a table where the griddle is now.
• Kitchen needs at least 1 bin. If you want to be good about it, add a prison disposals loop that goes into a small room. That small room should have a spaced (vacuumed) staging area that the trash gets shot out into with security-access windoors. Any prisoner should be able to shoot the junk out into space, but if any prisoner tries to escape via this method they'll just end up trapped in a vacuum chamber. Alternatively, do the same but without the vacuum and have security access on the shoot-junk-to-space-cannon-button.
• **(Optional)** Remodel the prison kitchen/cafe area because this is a remnant copy-paste from Citcode believe it or not. It's outdated, with all the new stuff TG added it can be made to look much prettier (graphically).

# Command
## Captain's Office
• Thinplating decal is on the carpet at the foot of the bed, yeet that.
## Captain's Mini-service
• Mini-bar area needs to be all access so visitors and non-HoP/Captain command staff can get drinks.
• Mini-bar needs **normal** drinks dispensers instead of `fullupgrade`.
• Either renovate the dark kitchen and integrate it into the Captain's office if you want that Star Trek Captain meme where the Captain is always a chef, or remove it. You will not get service willingly working in this place. It's a literal prison due to access anyway. Plus, service staff will roundly prefer to use their fully-stocked kitchen (and biogenerator) anyway.

# Misc
## Wrestling Arena
• Remove indestructible dungeon walls.
## Cargo
• Replace all instances of these with functional shelves, var-edit their color to brown if you must.
image

**Comment by W4rd3nn** [comment]
> Cargo
> • Replace all instances of these with functional shelves, var-edit their color to brown if you must.

ALL instances? As far as I know, these are intended to be primarily decorative, not functional. I can replace them with functional ones, but it ends up leaving the shelves bare. I can then go in and add items to them after but I don't think the quartermaster really needs like 8 different types of liquor. I agree the ones in the cargo bay should be functional, with maybe one or two decorative ones, but I don't think replacing all of them is very nice-looking or really useful.

**Comment by W4rd3nn** [comment]
it would be so awesome... it would be so cool...

**Comment by W4rd3nn** [comment]
map diff bot sux

**Comment by W4rd3nn** [comment]
STOP BREAKING REEEEEEE!!!!

**Comment by Thlumyn** [comment]
Small suggestion about virology; usually the virus food dispenser is within reach of the pandemic machine, which makes viro a lot more streamlined, perhaps switching its location with the intercom here (or the newscaster)?  (edit: additionally, viro doesn't seem to start with any plasma like it does on other maps)
image

**Comment by kittysmooch** [comment]
blueshield can't open their red sec locker in their little armory room.

**Comment by GoldenAlpharex** [comment]
This whole part of maintenance kind of just has barren plating, no wires or pipes or catwalks or anything. Compared to the rest, it feels on the empty side.
image

**Comment by Zergspower** [comment]
One major issue: you cant fix any turfs on floor one much like the original blueshift map. (a fix was made to the baseturfs later on)
image

Few minor things below mostly relating to areas
<details>
**Execution chamber areas are wonky**

image


**Maintenance areas here for Morgue maints also wonky, maybe have a door where that frog is and end the area there?**
image



**suggestion: have the area extend to cover those walls so it doesnt look so out of place compared to the rest of the walls around it that also jut out**
image

Some doors are with an area one way where the general room controls the door (power wise) while others have the maintenance room powering it. e.g the command room only needs the apc off next to it to be able to crawbar the door open vs the others around it being the internal apc needing to be off.

**looped cables**
**Bridge toolbay**
image

**Captains office**
image


**Looped and redundant cables in the silyl billy pen**
image

**Janitor closet waterpipe not connected**
image


**I'm not really up to date on security matters but the armory is armed a lot more lethally than i remember, even with 2 ions.**
image

**Since this is a SR map, we have polarized windows that will dim with a switch, would be neat to see those used here.**
image

**And the club room**
image

**Potentially the Robo room as well, which also needs some items added after the synth surgery update**
image


**Balance wise - I highly suggest not having this engineer here as it's a free engineering PDA/Headset and ID with modsuit**
image

**Disposals in Atmos is looping out - and while i really dig the uniqueness of it i feel they need a bit more workspace as large atmos projects need a fair bit of space between pipes to work and their workshop's smaller than chem**
image

</details>

**Comment by W4rd3nn** [comment]
> One major issue: you cant fix any turfs on floor one much like the original blueshift map. (a fix was made to the baseturfs later on) image
> 
> Few minor things below mostly relating to areas

Gwuh.. do you happen to know what the fix was? Or could link to the PR?

**Comment by Reco201** [comment]
Three missing wire spots
image
image
image

Weird spot of space you can go behind and cant build on 
image

Poppy the Possum is missing from engineering

And telecomms smes starts at zero power

**Comment by ChickyRooster** [comment]
Commdrobe might be missing, there's an autodrobe in the council chambers but that's about all I could find in the round, perhaps replace the Autodrobe there with a commdrobe?

**Comment by W4rd3nn** [comment]
wtf runtimestation...

**Comment by GoldenAlpharex** [comment]
<details>
<summary>There's a generic glitter grenade in the maints black market(?)</summary>

image

</details>

**Comment by GoldenAlpharex** [comment]
The Hidden Library (second floor, West) is only using symbol carpet variants, when it should be using regular carpet, as it looks really ugly otherwise.

**Comment by Zergspower** [comment]
funny small typo 
image

**Comment by W4rd3nn** [comment]
Yeah.. I still have to find and fix the "gardains" fucking everywhere

**Comment by Namwty** [comment]
Ordnenance related issues I've encountered:

*Freezer room air alarm is locked, this means you can't choose what scrubbers will take out of there and that you need AI or atmos tech for working with freezer.

*Tachyon-doppler isn't registering any bombs that blow up in bomb site or mayhaps bombs don't reach bombsite when mass driver is used, dunno. Regardless even maxcap wasn't detected by tachyon.

Minor stuff, which isn't a big deal, but just unpleasant consistency issue when compared to other maps. Like gas injectors being automatically on round start, or Oxygen big tank having absurd amounts of oxygen ( meaning if you mix something with oxygen you have to babysit it, as you might just empty secondary gas canister ). This can throw some folk off first time working on said lab, but other than that it's not a big deal.

**Comment by W4rd3nn** [comment]
GOLDEEEENNNNN WHYYYYYYYYY

**Comment by YakumoChen** [comment]
Its laggy, my client constantly had to buffer just to observe the station, and its definitely not my computer thats bad.

**Comment by ghost** [comment]
image
Disposal Bin outside cargo isn't actually hooked up to disposals

**Comment by W4rd3nn** [comment]
> 3. the floor markings here in dorms near the cryopods are incorrect (should be 1-2 rather than 2-3)

These are technically correct since Blueshift's lower deck was just walled off, not removed, canonically :3c 

Also that hole beneath CMO's office isn't part of the map originally. That was done in-round, my guess.

**Comment by Plejek** [comment]
Yello!
Was the consultant's bar drone removed, or did the (https://github.com/Skyrat-SS13/Skyrat-tg/pull/24507) pr do something that broke the drone spawner?

**Comment by dolphinemote** [comment]
Curator's backroom still has wide painting frame right one the floor.

**Review by GoldenAlpharex — APPROVED** [review]
Should be good to go once it passes the checks!

**Comment by W4rd3nn** [comment]
it compiles... now for the durability- i mean integration tests...

**Comment by W4rd3nn** [comment]
BRUH ITS ALREADY OUT OF DATE? HOW FUCKING FAST DO U FUCKS PR REEEEEEEEE

**Review by GoldenAlpharex — APPROVED** [review]
It's all good and ready to go!

**Pull request merged**
